lib/CircuitBreaker.js

```diff
@@ -1,10 +1,150 @@
+        // the topic and location, along with specifying if a probe should be
+        // used for transition and expiration.
+        // Query syntax:
+        // "with(topic=topicName,location=locationName,transition=true,expiration=true) and
+        //      PROM_METRIC_NAME{topic='${topic}',location='${location}'}"
+        // Note: should either specify the topic or the location, not both
+        // Behaviour:
+        // - no with clause + templated topic: creates a circuit breaker for each topic
+        // - with clause with a specific topic + templated topic: creates a circuit breaker for the specified topic
+        // - with clause with a specific location + templated location: creates a circuit breaker for the location
+        const withClause = probe.query.match(/with\((.*?)\)\sand\s/);
+
+        const withTopicTemplate = probe.query.includes('${topic}');
+        const withLocationTemplate = probe.query.includes('${location}');
+
+        const topic = withClause && withClause[1].match(/topic=(.*?)(,|$)/);
+        const location = withClause && withClause[1].match(/location=(.*?)(,|$)/);
+
+        // Ignore ambiguous configuration
+        if (
+            (topic && location) ||
+            (withTopicTemplate && withLocationTemplate) ||
+            (location && withTopicTemplate) ||
+            (topic && withLocationTemplate)
+        ) {
+            return;
+        }
+
+        const transition = withClause && withClause[1].match(/transition=(.*?)(,|$)/);
+        const expiration = withClause && withClause[1].match(/expiration=(.*?)(,|$)/);
+        // By default all probes are used for both transition and expiration
+        const isTransition = (transition && transition[1] === 'true') || !transition;
+        const isExpiration = (expiration && expiration[1] === 'true') || !expiration;
+
+        if (!isExpiration && !isTransition) {
+            return;
+        }
+
+        const query = withClause ? probe.query.replace(withClause[0], '') : probe.query;
+
+        // Consider a probe global (circuit breaks everything) when using a non templated query
+        if (!withClause && !withTopicTemplate && !withLocationTemplate) {
+            configs.global.push(probe);
+            return;
+        }
+
+        // When using the topic template, we need to create a probe for each relevant topic.
+        // In case the topic was specified, we need to replace the topic template with the actual topic
+        // If no topic template is used the probe is added as is to the topic probes
+        if (topic || withTopicTemplate) {
+            const topics = topic ? [topic[1]] : getAllTopics(lcConfig, repConfig, locations);
+            topics.forEach(topic => {
+                const probeConfig = {
+                    config: {
+                        ...probe,
+                    },
+                    transition: isTransition,
+                    expiration: isExpiration,
```

Comment:
I think it would be easier to read if the `transition` and `expiration` were renamed to better indicate that it means _"rule is enabled for transition"_